File: SistemaGestionTransporte/src/gestorAplicaciones/camion/CamionCisterna.java
package gestorAplicaciones.camion;

import java.util.ArrayList;

public class CamionCisterna extends Camion{

    public final static ArrayList<CamionCisterna> camiones = new ArrayList<CamionCisterna>();
    public CamionCisterna(String placa, String pais, double pesoMaximo, double capacidad) {
        super(placa, pais, pesoMaximo, capacidad);

    }

    public static ArrayList<CamionCisterna> getCamiones(){
        return CamionCisterna.camiones;
    }

    @Override
    public void calcularCostoCamion() {
        double factor = 0.005;
        double km = this.getRuta().get(this.getRuta().size()-1).getValue();
        this.setCosto(km * this.getCapacidad() * factor);
    }

    @Override
    public double valocidad() {
        double velocidadBase = 81.857;
        double factor = -0.392957;
        return velocidadBase + this.getCapacidad() * factor;
    }


    @Override
    public String ubicacion() {
        return null;
    }

    @Override
    public String tiempoRestante() {
        return null;
    }
}

File: python/SistemaGestionTransporte/src/gestorAplicaciones/Camion/CamionCisterna.py
from src.gestorAplicaciones.Camion.camion import Camion


class CamionCisterna(Camion):
    camiones = []

    def __init__(self, placa, pais, ciudadActual, pesoMaximo, capacidad):
        super().__init__(placa, pais, ciudadActual, pesoMaximo, capacidad)
        self.camiones.append(self)

    def calcularCostoCamion(self):
        factor = 0.005
        km = self.getRuta()[-1][1]
        self.setCosto(km * self.getCapacidad() * factor)

    def velocidad(self):
        velocidadBase = 81.857
        factor = -0.392957
        return velocidadBase + self.getCapacidad() * factor

    def __str__(self):
        return "\nTipo: Cisterna" \
               "\nPlaca: " + self.getPlaca() + \
            "\nPais: " + self.getPais() + \
            "\nCiudad actual: " + self.getCiudadActual() + \
            "\nPeso Maximo: " + str(self.getPesoMaximo()) + \
            "\nCapacidad: " + str(self.getCapacidad()) + \
            "\nDisponible: " + str(self.isDisponible()) + "\n"
